### PR TITLE
server/UI: fix some issues with network offering details

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes some issues with network offering details
(1) on UI: the serviceofferingid and maclearning is not passed to API when create network offering

(2) when create network offering, the details are not saved except internetprotocol, including
Promiscuous mode 
MAC address changes 
Forged transmits 
MAC learning 

(3) network offering details are not removed when delete a network offering.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [x] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 1]
```diff
@@ -6408,7 +6408,9 @@ public NetworkOfferingVO doInTransaction(final TransactionStatus status) {
                             detailsVO.add(new NetworkOfferingDetailsVO(offering.getId(), Detail.internetProtocol, String.valueOf(internetProtocol), true));
                         }
                         if (!detailsVO.isEmpty()) {
-                            networkOfferingDetailsDao.saveDetails(detailsVO);
```
Comment (weizhouapache) [inline server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:6411]: `saveDetails` will firstly remove all existing details. this causes issue (2) in the PR decription.

in my opinion. the issue is critical . @DaanHoogland @rohityadavcloud @shwstppr

**Comment by weizhouapache** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 1 above)
Comment (shwstppr) [reply server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:6411]: @weizhouapache this is used while creating a new offering so will it matter even if the code tries to delete existing details for the resourceId (service offering is here)?

Comment (weizhouapache) [reply server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:6411]: > @weizhouapache this is used while creating a new offering so will it matter even if the code tries to delete existing details for the resourceId (service offering is here)?

@shwstppr yes, there are some network offering details passed via API.
https://cloudstack.apache.org/api/apidocs-4.17/apis/createNetworkOffering.html

image


for example, the following details are not saved

Promiscuous mode
MAC address changes
Forged transmits
MAC learning

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 2]
```diff
@@ -6769,6 +6771,7 @@ public boolean deleteNetworkOffering(final DeleteNetworkOfferingCmd cmd) {
 
         annotationDao.removeByEntityType(AnnotationService.EntityType.NETWORK_OFFERING.name(), offering.getUuid());
         if (_networkOfferingDao.remove(offeringId)) {
+            networkOfferingDetailsDao.removeDetails(offeringId);
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:6774]: shouldn´t we first remove the details and than the oddering?

**Review by DaanHoogland — APPROVED** [review]
clgtm

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 2 above)
Comment (weizhouapache) [reply server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java]: thanks Daan.
fixed

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@shwstppr is this lgty?

**Review by shwstppr — APPROVED** [review]

**Pull request merged**